xray/plot/plot.py

```diff
@@ -456,11 +472,16 @@ def newplotfunc(darray, ax=None, xincrease=None, yincrease=None,
                                  vmax=cmap_params['vmax'],
                                  **kwargs)
 
-        ax.set_xlabel(xlab)
-        ax.set_ylabel(ylab)
+        # Label the plot with metadata
+        if add_labels:
+            ax.set_xlabel(xlab)
+            ax.set_ylabel(ylab)
+            ax.set_title(_title_for_slice(darray))
 
         if add_colorbar:
-            plt.colorbar(primitive, ax=ax, extend=cmap_params['extend'])
+            cbar = plt.colorbar(primitive, ax=ax, extend=cmap_params['extend'])
+            if darray.name and add_labels:
```

Comment:
since `DataArray.__init__()` defines the `name` attribute as `None` by default, probably better to use the `is not None` syntax:
```python
if darray.name is not None and add_labels:```